lib/Utilities/src/TaskProxy.hpp

```diff
@@ -100,29 +100,41 @@ class TaskProxy
     /**
      * Receive a item from the queue. The item will be copied, so enough space
      * must be provided!
-     * 
+     *
      * @param[out] item The item, which will be overwritten from the queue.
-     * 
+     *
      * @return If a item is successful received, it will return true otherwise false.
      */
     bool receive(T& item)
     {
         return m_queue.receive(&item, waitTimeTicks);
     }
 
+    /**
+     * Send a item to the front of the queue. The item will be copied!
+     * Attention: The copy takes place via memcpy, consider this in case of objects.
+     *
+     * @param[in] item  The item, which to send.
+     *
+     * @return If the item is successful sent, it will return true otherwise false.
+     */
+    bool sendToFront(T& item)

```

Comment:
Sending to front is ineffective, because all of the other items needs to be copied internal to the end, to be able to add one at the begin.